options.go

```diff
@@ -120,8 +121,29 @@ func HashRingConfig(c *hashring.Configuration) Option {
 // automatically.
 func Logger(l log.Logger) Option {
 	return func(r *Ringpop) error {
-		r.logger = l
-		r.log = l
+		if r.moduleLogger == nil {
+			r.moduleLogger = rutil.NewModuleLogger(l)
+		} else {
+			r.moduleLogger.SetLogger(rutil.RootModule, l)
+		}
+		return nil
+	}
+}
+
+// ModuleLevel is used to configure different log levels per module. Messages
+// with a level less than minLevel are silenced. By default, an unconfigured
+// module outputs all messages.
+func ModuleLevel(name string, minLevel rutil.Level) Option {
```

Comment:
I agree that making the config static makes it easy to see what and how can be changed.

We should include a `log.Logger` field in `LoggerConfig` for per module loggers and a `defaultLogger` to `LoggerConfiguration`.

Do you think it's worthwhile to keep `Logger`, `ModuleLevel` and `ModuleLogger` for one off updates at runtime?